src/spec.ts

```diff
@@ -101,8 +101,28 @@ export type Spec = UnitSpec | FacetSpec | LayerSpec;
 
 /* Custom type guards */
 
-export function isFacetSpec(spec: ExtendedSpec): spec is FacetSpec {
-  return spec['facet'] !== undefined;
+export function isFacetSpec(spec: ExtendedSpec | ExtendedFacetSpec): spec is FacetSpec {
```

Comment:
Distinction between facet spec and extended facet spec seems unnecessary.  Your code path for both cases are the same.  Plus, this change will become incorrect in the future.